src/components/search/data/searchLoader.test.jsx

```diff
@@ -124,4 +143,74 @@ describe('searchLoader', () => {
       }),
     );
   });
+
+  it('Redirect learners whose enterprise has enabled one academy.', async () => {
+    extractEnterpriseCustomer.mockResolvedValue(enterpriseCustomerFactory({ enable_one_academy: true }));
+    const academiesQuery = queryAcademiesList(mockEnterpriseCustomer.uuid);
+
+    when(mockQueryClient.ensureQueryData).calledWith(
+      expect.objectContaining({
+        queryKey: academiesQuery.queryKey,
+      }),
+    ).mockResolvedValue(mockAcademies);
+    ensureAuthenticatedUser.mockResolvedValue({ userId: 3, username: 'test-user' });
+
+    const mockParams = { enterpriseSlug: mockEnterpriseCustomer.slug };
+    const searchLoader = ({ request }) => makeSearchLoader(mockQueryClient)({ params: mockParams, request });
+
+    renderWithRouterProvider({
+      path: `/${mockEnterpriseCustomer.slug}/search`,
+      element: <div data-testid="root" />,
+      loader: searchLoader,
+    }, {
+      routes: [
+        {
+          path: `/${mockEnterpriseCustomer.slug}/academies/${mockAcademies[0].uuid}`,
+          element: <div data-testid="root" />,
```

Comment:
The `data-testid` for this route should be specific to the academy detail page, e.g. `<div data-testid="academy-detail" />`. The intent being that this test should assert that `data-testid="academy-detail"` exists after executing the `redirect` function in the loader. As is, there's no distinction between the search route element and the academy detail page route element.